### PR TITLE
🏗️ Maintenance/week 25: updates of testing, tooling and vulnerable libraries

### PR DESCRIPTION
**Pull request by pcrespov**
## What do these changes do?

- CHANGE: vulnerability upgrades of ``aiohttp`` and ``urllib3`` (reported by [dependabot](https://github.com/ITISFoundation/osparc-simcore/security/dependabot))
- CHANGE: upgrades all testing and tooling dependencies
- FIX: missing dependencies in sidecar requirement files
- CHANGE: ``scripts/codestyle.bash`` now installs all missing stub packages (e.g. types-PyYAML, types-aiofiles, etc)

NOTE: opened issue #2393 to keep a backlog of all features missing for requirements upgrade.

## Related issue/s

Weekly maintenance

## How to test

Affected all repo tests

### CODE REVIEW

File: packages/models-library/requirements/_tools.txt [hunk 1]
```diff
@@ -8,23 +8,23 @@ appdirs==1.4.4
     # via
     #   black
     #   virtualenv
-black==21.5b1
+black==21.6b0
```
Comment (colinRawlings) [inline packages/models-library/requirements/_tools.txt:11]: If you are bumping black do you need to submit also a formatting commit?

File: scripts/codestyle.bash [hunk 2]
```diff
@@ -31,6 +32,10 @@ ci() {
   echo "pylint"
   pylint --rcfile=.pylintrc "$BASE_PATH_DIR"/src/"$SRC_DIRECTORY_NAME" "$BASE_PATH_DIR"/tests
   echo "mypy"
+  # installing all missing stub packages (e.g. types-PyYAML, types-aiofiles, etc)
+  mkdir --parents "$CACHE_DIR"
+  mypy --cache-dir="$CACHE_DIR" --install-types
+  # runs mypy
   mypy --config-file mypy.ini --ignore-missing-imports "$BASE_PATH_DIR"/src/"$SRC_DIRECTORY_NAME" "$BASE_PATH_DIR"/tests
```
Comment (colinRawlings) [inline scripts/codestyle.bash:39]: should mypy.ini also have an absolute path (sorry I know this is not a change from this MR but it stood out)?

**Review by colinRawlings — APPROVED** [review]
Looks nice - just some small questions

File: packages/models-library/requirements/_tools.txt (hunk 1 above)
Comment (pcrespov) [reply packages/models-library/requirements/_tools.txt:11]: that is a good point, we could do this from now on. All black upgrades are followed by a re-formatting
I would propose this time to create a separate PR in which we black-format the entire repo with this version

File: scripts/codestyle.bash (hunk 2 above)
Comment (pcrespov) [reply scripts/codestyle.bash]: done

File: scripts/codestyle.bash [hunk 3]
```diff
@@ -28,10 +34,13 @@ ci() {
...
-  pylint --rcfile=.pylintrc "$BASE_PATH_DIR"/src/"$SRC_DIRECTORY_NAME" "$BASE_PATH_DIR"/tests
-  echo "mypy"
-  mypy --config-file mypy.ini --ignore-missing-imports "$BASE_PATH_DIR"/src/"$SRC_DIRECTORY_NAME" "$BASE_PATH_DIR"/tests
+  echo "pylint ..."
+  pylint --rcfile="$PYLINT_CONFIG" "$BASE_PATH_DIR"/src/"$SRC_DIRECTORY_NAME" "$BASE_PATH_DIR"/tests
+  echo "mypy ..."
+  # installing all missing stub packages (e.g. types-PyYAML, types-aiofiles, etc)
+  python3 -m pip install types-aiofiles types-PyYAML types-ujson
+  # runs mypy
+  mypy --config-file "$MYPY_CONFIG" --ignore-missing-imports "$BASE_PATH_DIR"/src/"$SRC_DIRECTORY_NAME" "$BASE_PATH_DIR"/tests
```
Comment (sanderegg) [inline scripts/codestyle.bash:43]: still don't get why we use 2 different versions of mypy... we should unify

**Review by sanderegg — APPROVED** [review]
we still use 2 different mypy, with probably even different versions. not sure why. anyway is out of the scope of that PR.

File: scripts/codestyle.bash (hunk 3 above)
Comment (pcrespov) [reply scripts/codestyle.bash:43]: Will make sure they are the same in a separate PR

**Pull request merged**